java-security/src/main/java/com/sap/cloud/security/servlet/XsuaaTokenAuthenticator.java

```diff
@@ -39,9 +64,83 @@ protected TokenAuthenticationResult authenticated(Token token) {
 		return TokenAuthenticatorResult.createAuthenticated(translatedScopes, token);
 	}
 
+    @Override
+    public TokenAuthenticationResult validateRequest(ServletRequest request, ServletResponse response) {
+        if (request instanceof HttpServletRequest && response instanceof HttpServletResponse) {
+            HttpServletRequest httpRequest = (HttpServletRequest) request;
+            String authorizationHeader = httpRequest.getHeader(HttpHeaders.AUTHORIZATION);
+            if (headerIsAvailable(authorizationHeader)) {
+                try {
+                    Token token = extractFromHeader(authorizationHeader);
+                    if (isIasXsuaaXchangeEnabled() && !isXsuaaToken(token)) {
```

Comment:
you can even put the `isIasXsuaaXchangeEnabled()` check to the first if-clause.

Reply:
@nenaraab , if I do that then it won't do any token verification in case ias to xsuaa token xchange is disabled